clippy_lints/src/manual_let_else.rs

```diff
@@ -85,6 +69,7 @@ impl<'tcx> LateLintPass<'tcx> for ManualLetElse {
                     if let Some(ident_map) = expr_simple_identity_map(local.pat, let_pat, if_then);
                     if let Some(if_else) = if_else;
                     if expr_diverges(cx, if_else);
+                    if pat_and_expr_can_be_question_mark(cx, let_pat, if_else).is_none();
```

Comment:
nit: As I said before (but, I ended up pointing at the wrong spot :sweat_smile:), I think this could check whether `question_mark` is allowed or not and still lint if it is. It's fine if it's not done, but I think it could make sense